libraries/abstractions/common_io/test/test_scripts/uart/test_iot_uart_test.py

```diff
@@ -0,0 +1,181 @@
+# Amazon FreeRTOS V1.2.3
+# Copyright (C) 2019 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+# http://aws.amazon.com/freertos
+# http://www.FreeRTOS.org
+
+
+import serial
+import csv
+import argparse
+import os, sys
+import re
+
+parentdir = os.path.dirname(os.getcwd())
+sys.path.insert(0, parentdir)
+from test_iot_test_template import test_template
+
+
+class uart_test(test_template):
+    """
+    Test class for uart tests.
+    """
+
+    outputfile = "./uart_res.txt"
+
+    def __init__(self, serial, ip, login, pwd, csv_handler):
+        self._func_list = [self.test_IotUartWriteReadAsync, self.test_IotUartBaudChange, self.test_IotUartWriteAsync]
+
+        self._serial = serial
+        self._ip = ip
+        self._login = login
+        self._pwd = pwd
+        self._cr = csv_handler
+
+    shell_script = "./test_iot_runonPI_uart.sh"
+
+    def test_IotUartWriteReadAsync(self):
+        """
+        Test body of uart write then read async test. Return pass if expected message is recieved
+        :return: 'Pass' or 'Fail'
+        """
+        dut_result = None
+        pi_result = None
+        dut_res_pattern = re.compile(r"AFQP_AssistedIotUARTWriteReadSync\W\sPASS")
+        pi_regex = re.compile(r"UART\sread-write")
+
+        self.run_shell_script(
+            " ".join([self.shell_script, self._ip, self._login, self._pwd, "0"])
+        )
+
+        self._serial.reset_input_buffer()
+
+        cmd = "iot_tests test 1 0"
+        self._serial.write("\r\n".encode("utf-8"))
+        self._serial.write(cmd.encode("utf-8"))
+        self._serial.write("\r\n".encode("utf-8"))
+
+        dut_res = str(self._serial.read(100))
+        dut_result = dut_res_pattern.search(dut_res)
+
```

Comment:
You may want to add ipc to synchronize between processes on rpi and dut. The latency of ssh can vary among locations. The output file may not ready at this point. I updated my tests with socket in this pr https://github.com/aws/amazon-freertos/pull/1356/files.

Reply:
Okay will take a look